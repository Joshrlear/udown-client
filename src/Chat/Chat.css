/* learn use mixins to make mobile responsive */
.chat_container {
  display: none;
  width: 100vw;
  height: 93vh;
  background: white;
}

.chat_container.active {
  display: block;
}

.chat_container .chat_title_container {
  width: 100%;
  height: 7vh;
  background-color: #6ab4ce;
}

.chat_container .chat_title_container .chat_title {
  display: block;
  height: 100%;
  padding-top: 1vh;
  text-align: center;
  margin-left: auto;
  margin-right: auto;
  color: white;
  font-size: 1.5em;
}

.chat_container .chat_title_container .back_arrow_container {
  display: block;
  position: absolute;
  top: 9vh;
  left: 2vh;
  height: 6vh;
}

.chat_container .chat_title_container .back_arrow_container a {
  display: inline-block;
  border-radius: 50%;
  text-decoration: none;
  color: white;
}

.chat_container .chat_title_container .back_arrow_container a:hover .back_arrow {
  border: 0.5em solid #1d2323;
}

.chat_container .chat_title_container .back_arrow_container a:hover .back_arrow:after {
  border-top: 0.5em solid #c8db16;
  border-right: 0.5em solid #c8db16;
}

.chat_container .chat_title_container .back_arrow_container .back_arrow {
  display: inline-block;
  width: 6vh;
  height: 6vh;
  border: 0.5vh solid white;
  border-radius: 50%;
  margin-right: 1.5em;
}

.chat_container .chat_title_container .back_arrow_container .back_arrow:after {
  content: '';
  margin-top: 1.05em;
  margin-left: 0.6em;
  width: 1.4em;
  height: 1.4em;
  border-top: 0.5vh solid white;
  border-right: 0.5vh solid white;
  -webkit-transform: rotate(-135deg);
  transform: rotate(-135deg);
}

a {
  cursor: pointer;
}

.chat_input_section_container {
  position: absolute;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  bottom: 1.5vh;
  width: 100vw;
  height: 38px;
}

.chat_input_section_container .input_container {
  width: 75%;
  height: 100%;
}

.chat_input_section_container .input_container input {
  -webkit-box-sizing: border-box;
          box-sizing: border-box;
  margin: 0px auto 0px auto;
  height: 100%;
}

.chat_input_section_container .button_container {
  width: 25%;
  height: 100%;
}

.chat_input_section_container .button_container .send_button {
  -webkit-box-sizing: border-box;
          box-sizing: border-box;
  width: 90%;
  height: 100%;
}
/*# sourceMappingURL=Chat.css.map */