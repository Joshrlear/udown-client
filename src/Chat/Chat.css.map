{
    "version": 3,
    "mappings": "AAEA,gDAAgD;AAQhD,AAAA,eAAe,CAAC;EACZ,OAAO,EAAE,IAAI;EACb,KAAK,EAAE,KAAK;EACZ,MAAM,EAAE,IAAW;EACnB,UAAU,EAAE,KAAK;CAkEoC;;AAtEzD,AAOI,eAPW,AAOV,OAAO,CAAC;EACL,OAAO,EAAE,KAAK;CAAG;;AARzB,AAWI,eAXW,CAWX,qBAAqB,CAAC;EAClB,KAAK,EAAE,IAAI;EACX,MAAM,EAAE,GAAG;EACX,gBAAgB,ECrBX,OAAO;CD6EmC;;AAtEvD,AAiBQ,eAjBO,CAWX,qBAAqB,CAMjB,WAAW,CAAC;EACR,OAAO,EAAE,KAAK;EACd,MAAM,EAAE,IAAI;EACZ,WAAW,EAAE,GAAG;EAChB,UAAU,EAAE,MAAM;EAClB,WAAW,EAAE,IAAI;EACjB,YAAY,EAAE,IAAI;EAClB,KAAK,EAAE,KAAK;EACZ,SAAS,EAAE,KAAK;CAAG;;AAzB/B,AA4BQ,eA5BO,CAWX,qBAAqB,CAiBjB,qBAAqB,CAAC;EAClB,OAAO,EAAE,KAAK;EACd,QAAQ,EAAE,QAAQ;EAClB,GAAG,EAAE,GAAG;EACR,IAAI,EAAE,GAAG;EACT,MAAM,EAAE,GAAG;CAqC8B;;AAtErD,AAoCY,eApCG,CAWX,qBAAqB,CAiBjB,qBAAqB,CAQjB,CAAC,CAAC;EACE,OAAO,EAAE,YAAY;EACrB,aAAa,EAAE,GAAG;EAClB,eAAe,EAAE,IAAI;EACrB,KAAK,EAAE,KAAK;CAQoC;;AAhDhE,AA2CgB,eA3CD,CAWX,qBAAqB,CAiBjB,qBAAqB,CAQjB,CAAC,AAOI,MAAM,CAAC,WAAW,CAAC;EAChB,MAAM,EAAE,KAAK,CAAC,KAAK,CCtDtB,OAAO;CDsDkC;;AA5C1D,AA8CgB,eA9CD,CAWX,qBAAqB,CAiBjB,qBAAqB,CAQjB,CAAC,AAUI,MAAM,CAAC,WAAW,AAAA,MAAM,CAAC;EACtB,UAAU,EAAE,KAAK,CAAC,KAAK,CCrD5B,OAAO;EDsDF,YAAY,EAAE,KAAK,CAAC,KAAK,CCtD9B,OAAO;CDsDwC;;AAhD9D,AAmDY,eAnDG,CAWX,qBAAqB,CAiBjB,qBAAqB,CAuBjB,WAAW,CAAC;EACR,OAAO,EAAE,YAAY;EACrB,KAAK,EAAE,GAAG;EACV,MAAM,EAAE,GAAG;EACX,MAAM,EAAE,iBAAiB;EACzB,aAAa,EAAE,GAAG;EAClB,YAAY,EAAE,KAAK;CAagB;;AAtEnD,AA4DgB,eA5DD,CAWX,qBAAqB,CAiBjB,qBAAqB,CAuBjB,WAAW,AASN,MAAM,CAAC;EACJ,OAAO,EAAE,EAAE;EACX,UAAU,EAAE,MAAM;EAClB,WAAW,EAAE,KAAK;EAClB,KAAK,EAAE,KAAK;EACZ,MAAM,EAAE,KAAK;EACb,UAAU,EAAE,iBAAiB;EAC7B,YAAY,EAAE,iBAAiB;EAC/B,cAAc,EAAE,eAAe;EAC/B,iBAAiB,EAAE,eAAe;EAClC,SAAS,EAAE,eAAe;CAAG;;AAIjD,AAAA,6BAA6B,CAAC;EAC1B,QAAQ,EAAE,QAAQ;EAClB,OAAO,EAAE,IAAI;EACb,MAAM,EAAE,KAAK;EACb,KAAK,EAAE,KAAK;EACZ,MAAM,EAAE,IAAI;CAsBe;;AA3B/B,AAQI,6BARyB,CAQzB,gBAAgB,CAAC;EACb,KAAK,EAAE,GAAG;EACV,MAAM,EAAE,IAAI;CAMS;;AAhB7B,AAaQ,6BAbqB,CAQzB,gBAAgB,CAKZ,KAAK,CAAC;EACF,UAAU,EAAE,UAAU;EACtB,MAAM,EAAE,iBAAiB;EACzB,MAAM,EAAE,IAAI;CAAG;;AAhB3B,AAmBI,6BAnByB,CAmBzB,iBAAiB,CAAC;EACd,KAAK,EAAE,GAAG;EACV,MAAM,EAAE,IAAI;CAMS;;AA3B7B,AAwBQ,6BAxBqB,CAmBzB,iBAAiB,CAKb,YAAY,CAAC;EACT,UAAU,EAAE,UAAU;EACtB,KAAK,EAAE,GAAG;EACV,MAAM,EAAE,IAAI;CAAG",
    "sources": [
        "Chat.sass",
        "../utilities/_variables.sass"
    ],
    "names": [],
    "file": "Chat.css"
}