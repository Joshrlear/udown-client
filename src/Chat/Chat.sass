@import "../utilities/variables"

/* learn use mixins to make mobile responsive */
@mixin screenSize()
    @if 100vw < 414px
        width: 414px
    @else
        width: 100vw


.chat_container
    display: none
    width: 100vw
    height: 100vh - 7vh
    background: white


    &.active
        display: block
    

    .chat_title_container
        width: 100%
        height: 7vh
        background-color: $blue-color


        .chat_title
            display: block
            height: 100%
            padding-top: 1vh
            text-align: center
            margin-left: auto
            margin-right: auto
            color: white
            font-size: 1.5em


        .back_arrow_container
            display: block
            position: absolute
            top: 9vh
            left: 2vh
            height: 6vh


            a
                display: inline-block
                border-radius: 50%
                text-decoration: none
                color: white


                &:hover .back_arrow
                    border: 0.5em solid $darkGrey-color

                &:hover .back_arrow:after
                    border-top: 0.5em solid $accent-color
                    border-right: 0.5em solid $accent-color


            .back_arrow
                display: inline-block
                width: 6vh
                height: 6vh
                border: 0.5vh solid white
                border-radius: 50%
                margin-right: 1.5em


                &:after
                    content: ''
                    margin-top: 1.05em
                    margin-left: 0.6em
                    width: 1.4em
                    height: 1.4em
                    border-top: 0.5vh solid white
                    border-right: 0.5vh solid white
                    -moz-transform: rotate(-135deg)
                    -webkit-transform: rotate(-135deg)
                    transform: rotate(-135deg)
            

a
    cursor: pointer


.chat_input_section_container
    position: absolute
    display: flex
    bottom: 1.5vh
    width: 100vw
    height: 38px


    .input_container
        width: 75%
        height: 100%


        input
            box-sizing: border-box
            margin: 0px auto 0px auto
            height: 100%


    .button_container
        width: 25%
        height: 100%


        .send_button
            box-sizing: border-box
            width: 90%
            height: 100%