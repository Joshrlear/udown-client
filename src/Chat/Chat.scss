@import "../utilities/variables";

@mixin desktop() {
    @media screen and ( min-width: 1100px ) {
      @content;
    }
  }
  
  @mixin tablet() {
    @media screen and ( min-width: 760px ) {
      @content;
    }
  }
  
  @mixin l-tablet() {
    @media screen and ( min-width: 900px ) {
      @content;
    }
  }

textarea, p {
    font-family: $poppins-font;
    font-size: 1.2rem;
}

.chat_container {
    display: none;
    position: absolute;
    z-index: 4;
    width: 100vw;
    height: 100vh;
    background: white;
    @include l-tablet() {
        width: 320px;
        height: 500px;
        bottom: 0px;
        left: 2vw;
        border: 2px lightgrey solid;
        border-bottom: none;
        border-radius: 10px 10px 0px 0px;
        box-sizing: initial;
    }


    &.active {
        display: block;
    }

    .chat_title_container {
        width: 100vw;
        height: 8vh;
        display: grid;
        grid-template-columns: repeat(3, 1fr);
        background-color: $blue-color;
        @include l-tablet() {
            width: inherit;
            height: 5vh;
            grid-template-columns: 80% 20%;
            border-radius: 7px 7px 0px 0px;
        }


        .chat_title {
            display: flex;
            justify-content: left;
            align-items: center;
            padding-left: 15px;
            height: 100%;
            text-align: center;
            color: white;
            font-size: 1.5rem;
            font-weight: 500;
            order: 0;
        }

        .back_arrow_container {
            display: block;
            height: 6vh;


            a {
                display: flex;
                position: relative;
                top: 2vh;
                left: 2vh;
                border-radius: 50%;
                text-decoration: none;
                color: white;

                &:hover .back_arrow {
                    border: 0.5em solid $darkGrey-color;
                }

                &:hover .back_arrow:after {
                    border-top: 0.5em solid $accent-color;
                    border-right: 0.5em solid $accent-color;
                }
            }
        }
    }
}

.chat_section {
    height: 80%;
    overflow: auto;
    @include l-tablet() {
        height: 77%;
    }

}

.message {
    padding: 10px 20px;

    h3 {
        font-size: 0.9rem;
        font-weight: 600;
    }

    p {
        font-size: 1.1rem;
        font-weight: 300;
    }
}

a {
    cursor: pointer;
}

.chat_input_section_container {
    position: fixed;
    display: grid;
    grid-template-columns: 72% 3% 20% 5%;
    bottom: 3vh;
    width: inherit;
    height: 38px;
    padding-top: 0px ;
    @include l-tablet() {
        width: inherit;
        bottom: 2vh;
    }

    div {
        padding-top: 0px;
        margin: 0% 0% 0% 8%;
    }

    .input_container {
        height: inherit;


        textarea {
            resize: none;
            box-sizing: border-box;
            border: 2px lightgrey solid;
            width: 100%;
            height: inherit;
            margin: 0px;

            &::placeholder {
                text-align: left;
                padding-left: 10px;
            }
        }
    }

    .button_container {
        width: 100%;
        height: 100%;
        justify-content: flex-end;


        .send_button {
            width: 100%;
            height: 100%;
            text-shadow: none;
            border: none;
            margin: 0px;
            font-size: 1rem;
            text-align: center;
            display: block;
            position: relative;
            cursor: pointer;
            border-radius: 20px;
            text-align: center;
            padding: 0px;
            text-decoration: none;
            color: white;
            font-weight: 600;
            background-color: $blue-color;
            transition: box-shadow 0.2s ease-out;
            &:hover {
              box-shadow: 0px 0px 100px $darkBlue-color inset;
            }
        }
    }
}

.close_btn {
    background-color: transparent;
    border: none;
    width: 100%;
    display: grid;
    grid-template-rows: repeat(3, 1fr);
    grid-template-columns: repeat(5, 1fr);
    height: inherit;
    text-decoration: none;
    order: 1;
}

.fas {
    display: flex;
    justify-content: flex-end;
    grid-row: 2/3;
    grid-column: 2/3;
    margin-right: 5vw;
    color: white;
    @include desktop() {
        display: initial;
        justify-content: initial;
        margin: 0px;
    }
}
