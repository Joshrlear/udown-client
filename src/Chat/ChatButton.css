.chat_button_container {
  position: fixed;
  width: 50px;
  height: 50px;
  bottom: 15px;
  right: 15px;
  z-index: 3;
}

.chat_button {
  display: none;
  background-color: #6ab4ce;
  border-radius: 50%;
  border-radius: 50%;
  padding-top: 6px;
  text-align: center;
  color: white;
  width: 0px;
  height: 0px;
  -webkit-box-shadow: 5px 5px 4px rgba(64, 166, 255, 0.3) inset, -3px -3px 5px rgba(145, 255, 112, 0.3) inset;
          box-shadow: 5px 5px 4px rgba(64, 166, 255, 0.3) inset, -3px -3px 5px rgba(145, 255, 112, 0.3) inset;
  text-shadow: -1px 1px 1px rgba(0, 0, 0, 0.4);
  -webkit-transition: -webkit-box-shadow 0.2s ease-out;
  transition: -webkit-box-shadow 0.2s ease-out;
  transition: box-shadow 0.2s ease-out;
  transition: box-shadow 0.2s ease-out, -webkit-box-shadow 0.2s ease-out;
}

.chat_button h2 {
  font-size: 15px;
}

.chat_button.active {
  display: block;
  width: 50px;
  height: 50px;
}

.chat_button:hover {
  cursor: pointer;
  -webkit-box-shadow: 0px 0px 100px rgba(158, 239, 70, 0.5) inset, 0px 0px 100px rgba(158, 239, 70, 0.5) inset;
          box-shadow: 0px 0px 100px rgba(158, 239, 70, 0.5) inset, 0px 0px 100px rgba(158, 239, 70, 0.5) inset;
}

.chat_button:hover.opened {
  width: 150%;
  height: 150%;
  bottom: -50%;
  right: -50%;
  border-radius: 0%;
  z-index: 3;
}
/*# sourceMappingURL=ChatButton.css.map */