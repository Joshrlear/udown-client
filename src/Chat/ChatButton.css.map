{
    "version": 3,
    "mappings": "AAEA,AAAA,sBAAsB,CAAC;EACnB,QAAQ,EAAE,KAAK;EACf,KAAK,EAAE,IAAI;EACX,MAAM,EAAE,IAAI;EACZ,MAAM,EAAE,IAAI;EACZ,KAAK,EAAE,IAAI;EACX,OAAO,EAAE,CAAC;CAGS;;AATvB,AAQI,sBARkB,AAQjB,OAAO,CAAC;EACL,OAAO,EAAE,CAAC;CAAG;;AAGrB,AAAA,YAAY,CAAC;EACT,OAAO,EAAE,IAAI;EACb,gBAAgB,ECbP,OAAO;EDchB,aAAa,EAAE,GAAG;EAClB,aAAa,EAAE,GAAG;EAClB,WAAW,EAAE,GAAG;EAChB,UAAU,EAAE,MAAM;EAClB,KAAK,EAAE,KAAK;EACZ,KAAK,EAAE,GAAG;EACV,MAAM,EAAE,GAAG;EACX,UAAU,EAAE,GAAG,CAAC,GAAG,CAAC,GAAG,CAAC,uBAAoB,CAAC,KAAK,EAAG,IAAG,CAAE,IAAG,CAAC,GAAG,CAAC,wBAAqB,CAAC,KAAK;EAC7F,WAAW,EAAG,IAAG,CAAC,GAAG,CAAC,GAAG,CAAC,kBAAkB;EAC5C,UAAU,EAAE,wBAAwB;EACpC,OAAO,EAAE,CAAC;CAoBe;;AAjC7B,AAeI,YAfQ,CAeR,EAAE,CAAC;EACC,SAAS,EAAE,IAAI;CAAG;;AAhB1B,AAkBI,YAlBQ,AAkBP,OAAO,CAAC;EACL,OAAO,EAAE,KAAK;EACd,KAAK,EAAE,IAAI;EACX,MAAM,EAAE,IAAI;CAAG;;AArBvB,AAuBI,YAvBQ,AAuBP,MAAM,CAAC;EACJ,MAAM,EAAE,OAAO;EACf,UAAU,EAAE,GAAG,CAAC,GAAG,CAAC,KAAK,CAAC,uBAAuB,CAAC,KAAK,EAAE,GAAG,CAAC,GAAG,CAAC,KAAK,CAAC,uBAAuB,CAAC,KAAK;CAQjF;;AAjC3B,AA2BQ,YA3BI,AAuBP,MAAM,AAIF,OAAO,CAAC;EACL,KAAK,EAAE,IAAI;EACX,MAAM,EAAG,IAAI;EACb,MAAM,EAAE,IAAI;EACZ,KAAK,EAAE,IAAI;EACX,aAAa,EAAE,EAAE;EACjB,OAAO,EAAE,CAAC;CAAG",
    "sources": [
        "ChatButton.sass",
        "../utilities/_variables.sass"
    ],
    "names": [],
    "file": "ChatButton.css"
}