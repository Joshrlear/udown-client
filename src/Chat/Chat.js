import React, { Component } from 'react'
import io from 'socket.io-client'
import config from '../config'
import UdownContext from '../UdownContext'
import ScrollToBottom from 'react-scroll-to-bottom'
import './Chat.scss'

let socket

export default class Chat extends Component {
    constructor(props) {
        super(props);
        this.chatInput = React.createRef()
        this.chatSection = React.createRef()
        this.messagesEnd = React.createRef()
        this.state = {
            currentMessage: '',
            messages: [],
            userTyping: [],
        }
    }

    static contextType = UdownContext

    componentWillMount() {
        // connect to socket if no socket connected
        !socket && (this.connectSocket())
    }

    componentDidUpdate() {
        this.messagesEnd.current && this.messagesEnd.current.scrollIntoView({ behavior: "smooth" })
    }

    connectSocket() {
        socket = io(config.API_ENDPOINT)
        socket.on('chat_message', msg => {
            this.setState({
                messages: [
                    ...this.state.messages, 
                    msg
                ]
            })
        })

        // displays: username is typing...
        socket.on('typing', user => {
            let timeout = null
            // if user not found in state
            if (this.state.userTyping.includes(user)) {
                // clear any previous timeout
                clearTimeout(timeout)
                // begin timeout
                timeout = setTimeout(() => {
                    
                    // after 3 seconds all usernames will be removed
                    this.setState({
                        userTyping: []
                    })
                }, 3000)
            }
            else {
                this.setState({
                    userTyping: [...this.state.userTyping, user]
                })
            }
        })
    }

    closeChat = e => {
        e.preventDefault()
        this.context.closeChat()
    }

    handleInput = (message) => {
        const user = localStorage.username
        
        socket.emit('typing', user)
        const messageInfo = {
            username: localStorage.username,
            message: message
        }

        this.setState({
            currentMessage: messageInfo
        })
    }

    sendMessage= e => {
        e.preventDefault()
        const newMsg = this.state.currentMessage
        const { username, message } = this.state.currentMessage
        
        this.setState({
            messages: [...this.state.messages, newMsg]
        })
        socket.emit('chat_message', ({ username: username, message: message}))
        this.chatInput.current.value = ''
    }

    render() {
        const chatToggle = !this.context.chatOpened ? 'chat_container' : 'chat_container active'
        return(
        <>
            <div className={ chatToggle }>
                <div className="chat_title_container">
                
                    <button id="close_btn" className="close_btn" href="javascript:void(0)" onClick={e => this.closeChat(e)}><i className="fas fa-times fa-2x"></i></button>
                    <h1 className="chat_title">Chat</h1>
                </div>
                <ScrollToBottom className="chat_section">
                    <ul className="messages">
                        {this.state.messages.map((msg, i) => <li key={i} className="message">
                            <div  className="message_container">
                                <h3 className="from">{`${msg.username}:`}</h3>
                                <p className="message_text">{msg.message}</p>
                            </div>
                        </li>)}
                    </ul>
                    <ul className="users_typing">
                        {this.state.userTyping.map((user, i) => <li key={i} className="user_typing">
                            <p className="typing_msg"><em>{`${user} is typing...`}</em></p>
                        </li>)}
                    </ul>
                </ScrollToBottom>
                <div className="forScroll" ref={ this.messagesEnd }/>
                <form className="chat_input_section_container"
                onSubmit={e => {this.sendMessage(e)}}>
                    <div className="input_container">
                        <textarea
                            ref={ this.chatInput } 
                            className="chat_intput" 
                            placeholder="type message..."
                            onChange={e => this.handleInput(e.target.value)}/>
                    </div>
                    <br/>
                    <div className="button_container">
                        <button 
                            className="send_button" 
                            type="submit">
                                Send
                        </button>
                    </div>
                    
                </form>
            </div>
            </>
    )
    }
}