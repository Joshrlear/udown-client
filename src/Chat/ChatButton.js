import React, { Component, useContext } from 'react'
import { Link } from 'react-router-dom';
import UdownContext from '../UdownContext'
import ChatContext from './ChatContext'
import './ChatButton.sass'

export default class ChatButton extends Component {

    static contextType = UdownContext

    // click opens chat and removes nav
    handleClick = (openMsg, navStatus, navOff, navOn) => {

        openMsg()
        // if nav class on turn off after 1 sec
        setTimeout(() => 
            (navStatus === 'nav_on') 
            ? navOff() 
            : navOn()
            , 1000
        )
    }

    render() {
        const buttonClass = !this.context.chatOpened ? "chat_button_container" : "chat_button_container opened"

        return (
            <ChatContext.Consumer>
                { (navStatus, navOff, navOn) => (
                    <UdownContext.Consumer>
                        {(openMsg, msgBtnClass) => (
                            <div className="chat_button_container">
                                <Link to="/chat">
                                    <button 
                                        className={msgBtnClass} 
                                        /* onClick={this.handleClick(openMsg, navStatus, navOff, navOn)} */>
                                    </button>
                                </Link>
                            </div>
                        )}
                    </UdownContext.Consumer>
                )}
            </ChatContext.Consumer>
        );
    } 
}