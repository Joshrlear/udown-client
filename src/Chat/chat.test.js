import React from 'react';
import ReactDOM from 'react-dom';
import { shallow, mount, render } from 'enzyme'
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json'
import { BrowserRouter } from 'react-router-dom';
import Chat from './Chat';

configure({ adapter: new Adapter() })

describe('Chat', () => {
    const mappedComponents = ['Chat']
    mappedComponents.map(Component => {
        it(`renders ${Component} without crashing`, () => {
            const div = document.createElement('div');
            ReactDOM.render(
            <BrowserRouter>
               { `<${Component }/>` }
            </BrowserRouter>, div);
            ReactDOM.unmountComponentAtNode(div);
        })
        
    })
    
    it(`handleInput should update state.currentMessage`, () => {
    const wrapper = shallow(<Chat />).instance();
    wrapper.handleInput('message')
    expect(wrapper.state.currentMessage).toStrictEqual({
        'username': undefined,
        'message': 'message'
    })
  })
})