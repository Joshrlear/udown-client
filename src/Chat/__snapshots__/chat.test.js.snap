// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chat renders Chat correctly 1`] = `
<Fragment>
  <div
    className="chat_container"
  >
    <div
      className="chat_title_container"
    >
      <button
        className="close_btn"
        href="#"
        id="close_btn"
        onClick={[Function]}
      >
        <i
          className="fas fa-times fa-2x"
        />
      </button>
      <h1
        className="chat_title"
      >
        Chat
      </h1>
    </div>
    <_default
      className="chat_section"
    >
      <ul
        className="messages"
      />
      <ul
        className="users_typing"
      />
    </_default>
    <div
      className="forScroll"
    />
    <form
      className="chat_input_section_container"
      onSubmit={[Function]}
    >
      <div
        className="input_container"
      >
        <textarea
          className="chat_intput"
          onChange={[Function]}
          placeholder="type message..."
        />
      </div>
      <br />
      <div
        className="button_container"
      >
        <button
          className="send_button"
          type="submit"
        >
          Send
        </button>
      </div>
    </form>
  </div>
</Fragment>
`;
